### DISCUSSION TITLE
Are you supposed to / allowed to create more than one RenderPass in a single draw call?

### DISCUSSION THREAD

**Discussion by laptou** [gfx-rs/wgpu#2368] [Q&A] [answered]
I have a graph-based renderer that works by rendering shaders into intermediate textures, and then using those textures as inputs to the next group of shaders. I created a single `RenderPass` for each shader, and I seemed to get away with this when using Vulkan on Android and Linux. However, it doesn't seem to be working on Metal on iOS -- am I not supposed to do this?

**Comment by laptou** [comment, marked as answer]
Actually, since I'm rendering to a bunch of different textures, it stands to reason that I have to create multiple `RenderPass`es. So then am I only supposed to create one per `CommandEncoder`? Because I notice that when I do a GPU trace using Xcode, only the first `RenderPass` shows up, and I get a black screen :(

**Reply by kvark** [reply]
wgpu is a safe and portable API. If you were doing something you weren't expected to do, you'd get a validation error.
If you see something not working on a particular platform, it's likely an issue. Are you using `store: false` by any chance in your render passes? Please file a bug otherwise.

**Reply by laptou** [reply]
Oops, the issue I was seeing with the black screen was actually not related to `wgpu`.
> If you were doing something you weren't expected to do, you'd get a validation error.

Got it, I guess that answers my question.
